spec.bs

```diff
@@ -6,57 +6,417 @@ Status: CG-DRAFT
 
-  Promise<any> run(DOMString name, optional SharedStorageRunOperationMethodOptions options = {});
+    Promise<any> set(DOMString key,
+                     DOMString value,
+                     optional SharedStorageSetMethodOptions options = {});
+
+    Promise<any> append(DOMString key,
+                        DOMString value);
+
+    Promise<any> delete(DOMString key);
+
+    Promise<any> clear();
+
+    [Exposed=(SharedStorageWorklet)] Promise<DOMString> get(DOMString key);
+
+    [Exposed=(SharedStorageWorklet)] Promise<unsigned long> length();
+
+    [Exposed=(SharedStorageWorklet)] Promise<double> remainingBudget();
 
-  readonly attribute SharedStorageWorklet worklet;
-};
+    [Exposed=(SharedStorageWorklet)] SharedStorageIterator keys();
 
-[Exposed=(Window)]
-interface SharedStorageWorklet {
-  Promise<undefined> addModule(USVString moduleURL);
-};
+    [Exposed=(SharedStorageWorklet)] SharedStorageIterator entries();
 
-dictionary SharedStorageSetMethodOptions {
-  boolean ignoreIfPresent;
-};
+    [Exposed=(Window)] readonly attribute SharedStorageWorklet worklet;
+  };
 
-dictionary SharedStorageRunOperationMethodOptions {
-  object data;
-};
+  dictionary SharedStorageSetMethodOptions {
+    boolean ignoreIfPresent;
+  };
 
-dictionary SharedStorageUrlWithMetadata {
-  required USVString url;
-  object reportingMetadata;
-};
+  dictionary SharedStorageRunOperationMethodOptions {
+    object data;
+  };
+
+  enum IteratorMode { "key", "keyValue" };
+
+  [Exposed=(SharedStorageWorklet)]
+  interface SharedStorageIterator {
+    constructor(IteratorMode mode);
+    Promise<object> next();
+  };
 </xmp>
+
+  ## Note on Parallelism ## {#parallelism}
+
+  The {{SharedStorage}} interface runs its methods <a href="https://html.spec.whatwg.org/multipage/infrastructure.html#in-parallel">in parallel</a>. To avoid race conditions within a single context, each context that calls into the interface will have its own <a href="https://html.spec.whatwg.org/multipage/infrastructure.html#parallel-queue">parallel queue</a> for handling operations.
+
+  In particular, the {{Window}}'s {{SharedStorage}} operations for a given frame will use a separate <a href="https://html.spec.whatwg.org/multipage/infrastructure.html#parallel-queue">parallel queue</a> from that used by any {{SharedStorageWorklet}} that may be spun up by this frame.
+
+  When an algorithm before asks for steps to be enqueued, these steps should be enqueued on the <a href="https://html.spec.whatwg.org/multipage/infrastructure.html#parallel-queue">parallel queue</a> of the calling context, unless otherwise specified.
+
+  ## Run Operation Methods ## {#run-op}
+
+  <div algorithm>
+    The <dfn method for="SharedStorage">run(|name|, |options|)</dfn> method steps are:
+
+    1. Let |promise| be a new promise.
+    1. <a href="https://html.spec.whatwg.org/multipage/infrastructure.html#enqueue-the-following-steps">Enqueue the following steps</a>:
```

Comment:
I think its prefered to write something more like `Run the following steps [=in parallel=]:`.

Reply:
Done.